src/box/raft.c

```diff
@@ -129,8 +129,11 @@ box_raft_update_synchro_queue(struct raft *raft)
 			struct error *err = diag_last_error(diag_get());
 			errcode = box_error_code(err);
 			diag_log();
+			if (errcode == ER_IN_ANOTHER_PROMOTE)
+				fiber_sleep(0);
```

Comment:
This needs either a comment why is the sleep done only for "another promote", or it needs to be done unconditionally. To be frank I see nothing wrong with just making it unconditional.

Reply:
Let's do this unconditionally, but only if the loop has to repeat again.
```diff
--- a/src/box/raft.c
+++ b/src/box/raft.c
@@ -123,14 +123,21 @@ box_raft_update_synchro_queue(struct raft *raft)
                return;
        int rc = 0;
        uint32_t errcode = 0;
+       bool try_again = false;
        do {
                rc = box_promote_qsync();
                if (rc != 0) {
                        struct error *err = diag_last_error(diag_get());
                        errcode = box_error_code(err);
                        diag_log();
+                       if (!fiber_is_cancelled() &&
+                           (errcode == ER_QUORUM_WAIT ||
+                           errcode == ER_IN_ANOTHER_PROMOTE)) {
+                               try_again = true;
+                               fiber_sleep(0);
+                       }
                }
-       } while (rc != 0 && errcode == ER_QUORUM_WAIT && !fiber_is_cancelled());
+       } while (try_again);
 }
```